sql/core/src/main/scala/org/apache/spark/status/api/v1/sql/SqlResource.scala

```diff
@@ -84,18 +130,52 @@ private[v1] class SqlResource extends BaseAppResource {
     }
 
     val duration = exec.completionTime.getOrElse(new Date()).getTime - exec.submissionTime
-    val planDetails = if (details) exec.physicalPlanDescription else ""
-    val metrics = if (details) printableMetrics(exec.metrics, exec.metricValues) else Seq.empty
+    val planDetails = if (planDescription) exec.physicalPlanDescription else ""
+    val nodes =
+      if (details) {
+        printableMetrics(
+          sparkPlanGraphNodes, exec.metricValues, nodeIdAndWSCGIdMap)
+      } else {
+        Seq.empty
+      }
+    val edges = if (details) sparkPlanGraphEdges else Seq.empty
+
     new ExecutionData(
       exec.executionId,
       status,
       exec.description,
       planDetails,
-      metrics,
       new Date(exec.submissionTime),
       duration,
       running,
       completed,
-      failed)
+      failed,
+      nodes,
+      edges)
   }
+
+  private def printableMetrics(sparkPlanGraphNodes: Seq[SparkPlanGraphNode],
+    metricValues: Map[Long, String],
+    nodeIdAndWSCGIdMap: Map[Long, Option[Long]]): Seq[Node] = {
+
+    def getMetric(metricValues: Map[Long, String], accumulatorId: Long,
+      metricName: String): Option[Metric] = {
+
+      metricValues.get(accumulatorId).map( mv => {
+        val metricValue = if (mv.startsWith("\n")) mv.substring(1, mv.length) else mv
+        Metric(metricName, metricValue)
```

Comment:
Actually after https://github.com/apache/spark/pull/28037, part of the metrics name is in the value...
I can see the json output like:
```
{
"nodeId": 9,
"nodeName": "WholeStageCodegen (1)",
"metrics": [
{
"name": "duration",
"value": "total (min, med, max (stageId: taskId))\n4.3 s (269 ms, 270 ms, 272 ms (stage 3.0: task 16))"
}
]
}
```
But this is not strongly related to this PR. We can just fix it in another PR.

Reply:
Yes, sounds good.